### DISCUSSION TITLE
Your question

### DISCUSSION THREAD

**Discussion by tyfbyhs** [aclap-dev/video-downloadhelper#509] [Help and Q&A]
### Your question

Hi. I'm not able to download video. I copied the information.

Thanks.
version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: zh-CN
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "i686",
    "node": "10"
  },
  "home": "C:\\Users\\19660",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

### Addon & CoApp version and details

![QQ截图20240519174225](https://github.com/aclap-dev/video-downloadhelper/assets/170161166/4b54ea82-5c95-4f5d-bbf8-684bb3f6611a)

Exit code: 1

moz-extension://ae78d362-bf4d-4133-9bdd-9ca6c996291e/background/main.js:1
receive@moz-extension://ae78d362-bf4d-4133-9bdd-9ca6c996291e/background/main.js:1:13157
nd</callCatchAppNotFound/</<@moz-extension://ae78d362-bf4d-4133-9bdd-9ca6c996291e/background/main.js:1:42626

**Comment by GrampaWildWilly** [comment]
We need the URL of the web page on which your video is hosted.

**Reply by tyfbyhs** [reply]
live.fc2.com   （guangdongvideo.com）  and   youtube.com

**Reply by paulrouget** [reply]
Do you use an antivirus? Sometimes antivirus block the ffmpeg binary.

**Reply by tyfbyhs** [reply]
My computer don't have antivirus

**Reply by tyfbyhs** [reply]
![K2{5()YKIBSZ EI95(7L%2](https://github.com/aclap-dev/video-downloadhelper/assets/170161166/e47becd9-2be2-4e67-942b-c901ebe353b6)

**Reply by tyfbyhs** [reply]
what should i do

**Reply by paulrouget** [reply]
Do you use a proxy?

**Reply by GrampaWildWilly** [reply]
You have supplied URLs of web sites.  That's not good enough.  We need URLs of specific videos.

**Reply by tyfbyhs** [reply]
I use it, but I have never had this problem before

**Reply by tyfbyhs** [reply]
I can't download any videos from these two websites

**Comment by Mattmjs** [comment]
Here is a random link: https://live.fc2.com/87158828/ , I made the important part bold which may be the problem.

ffprobe version 6.1-essentials_build-www.gyan.dev Copyright (c) 2007-2023 the FFmpeg developers
  built with gcc 12.2.0 (Rev10, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --pkg-config=pkgconf --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-dxva2 --enable-d3d11va --enable-libvpl --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      58. 29.100 / 58. 29.100
  libavcodec     60. 31.102 / 60. 31.102
  libavformat    60. 16.100 / 60. 16.100
  libavdevice    60.  3.100 / 60.  3.100
  libavfilter     9. 12.100 /  9. 12.100
  libswscale      7.  5.100 /  7.  5.100
  libswresample   4. 12.100 /  4. 12.100
  libpostproc    57.  3.100 / 57.  3.100
mime type is not rfc8216 compliant
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/42/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/12/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/32/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/52/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/22/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/42/data/77100.ts?time=1716273357&hash=893e1692e5251902b59fac1bcbbbbbb49a00eb515760857acf0efcd03b1327e1' for reading
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/42/data/77101.ts?time=1716273358&hash=070c5b9d96e49af39257a1c5955078f6f5e58c7fc0485a7b632d20e03f99e3fe' for reading
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/12/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/12/data/77137.ts?time=1716273356&hash=6876cc4a6caceed29ceb74da2c4b69be84861a4783ec3084491a4bc55703e9bd' for reading
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/12/data/77138.ts?time=1716273357&hash=b545905892172176ce471919e16128b3d2fba455da0e4ccae3eae35da09ff083' for reading
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/32/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/32/data/77137.ts?time=1716273356&hash=d324c98708fdb0ec756616201677c36ba06fda865733809057890d3bc19aaaa5' for reading
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/32/data/77138.ts?time=1716273357&hash=bdadbfaf55ae0c3df410d3791748b8a52e80251111f3adacbc6bc94e5581e905' for reading
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/52/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76912.ts?time=1716273357&hash=7ab4fa125952e39f7db5410e0a2ed94c1495bfb3f6898d4e96fe18ad0f2f3dc7' for reading
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76913.ts?time=1716273358&hash=a8473c6ff4c9995c87870faa3b8d82fcd68d744e31a6a969e9099b9abca0c77b' for reading
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/22/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/22/data/77142.ts?time=1716273358&hash=07cf0f98c38b2f1334bd38a721f40d82030b4358f83fa7391376035eeb5a8169' for reading
[hls @ 00000261403bafc0] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/22/data/77143.ts?time=1716273359&hash=658df552ba6c1657617287a4cb6ab6db72a8754d1e8db46a24e45c52e083e4e9' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922292502).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922288748).
[https @ 00000261408f5e00] Opening 'https://us-west-1-media.live.fc2.com/a/stream/87158828/52/playlist?c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj' for reading
[hls @ 00000261403bafc0] Skip ('#EXT-X-VERSION:3')
[https @ 0000026141579540] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76914.ts?time=1716273359&hash=b94cc1f07a3520e7b2b65ac61d89c2fa00db670806dd5bec00e4492522851a63' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922382592).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922378838).
[https @ 00000261425d0e80] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76915.ts?time=1716273360&hash=7aac4f76c892b1e38464031ef1505d0108127f7e1c631ec162c11d964c628055' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922427637).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922423883).
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922472682).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922468928).
[https @ 0000026141579540] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76916.ts?time=1716273361&hash=c658a5b78570e208aefa09a5948bc84bd15dc059d18dfd3a48931152c38c0dcf' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922517727).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922513973).
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922562772).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922559018).
[https @ 00000261425d0e80] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76917.ts?time=1716273362&hash=7c32ea4cc5a40f25fefc0e034ce7a58549aaa26a4a4240bf8df712be3159a5be' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922607817).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922604063).
[https @ 0000026141579540] Opening 'https://us-west-1-media-worker1100.live.fc2.com/a/stream/v1/87158828/52/data/76918.ts?time=1716273363&hash=b25dbc38955acf290ed90ea31efef5fafe20e7dc3b754d7702b3980c4d203b1a' for reading
[mpegts @ 000002614100d900] Packet corrupt (stream = 0, dts = 6922697907).
[hls @ 00000261403bafc0] Packet corrupt (stream = 6, dts = 6922694153).
**[hls @ 00000261403bafc0] Could not find codec parameters for stream 0 (Video: h264 ([27][0][0][0] / 0x001B), none): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 1 (Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp): unspecified sample rate
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 2 (Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp): unspecified sample rate
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 3 (Video: h264 ([27][0][0][0] / 0x001B), none): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 4 (Video: h264 ([27][0][0][0] / 0x001B), none): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 5 (Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp): unspecified sample rate
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 8 (Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp): unspecified sample rate
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[hls @ 00000261403bafc0] Could not find codec parameters for stream 9 (Video: h264 ([27][0][0][0] / 0x001B), none): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options**
Input #0, hls, from 'https://us-west-1-media.live.fc2.com/a/stream/87158828/2/master_playlist?targets=10,20,30,40,50,90&c=Jb8dOkblboUF_m-spn_wI&d=JPATQaKoN4pYZliWUOntKqORW6UWag_TPoglLZJSbYEpx9F7h0_Q6w7NupUAgPJj':
  Duration: N/A, start: 76913.911756, bitrate: N/A
  Program 0
    Metadata:
      variant_bitrate : 3600000
  Stream #0:0: Video: h264 ([27][0][0][0] / 0x001B), none, 90k tbr, 90k tbn
    Metadata:
      variant_bitrate : 3600000
  Stream #0:1: Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp
    Metadata:
      variant_bitrate : 3600000
  Program 1
    Metadata:
      variant_bitrate : 270000
  Stream #0:2: Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp
    Metadata:
      variant_bitrate : 270000
  Stream #0:3: Video: h264 ([27][0][0][0] / 0x001B), none, 90k tbr, 90k tbn
    Metadata:
      variant_bitrate : 270000
  Program 2
    Metadata:
      variant_bitrate : 2160000
  Stream #0:4: Video: h264 ([27][0][0][0] / 0x001B), none, 90k tbr, 90k tbn
    Metadata:
      variant_bitrate : 2160000
  Stream #0:5: Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp
    Metadata:
      variant_bitrate : 2160000
  Program 3
    Metadata:
      variant_bitrate : 5400000
  Stream #0:6: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 90k tbn
    Metadata:
      variant_bitrate : 5400000
  Stream #0:7: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 5400000
  Program 4
    Metadata:
      variant_bitrate : 720000
  Stream #0:8: Audio: aac ([15][0][0][0] / 0x000F), 0 channels, fltp
    Metadata:
      variant_bitrate : 720000
  Stream #0:9: Video: h264 ([27][0][0][0] / 0x001B), none, 90k tbr, 90k tbn
    Metadata:
      variant_bitrate : 720000

**Reply by paulrouget** [reply]
We are supposed to be using the proper values for `analyzeduration`. Did you run ffmpeg manually?

**Reply by tyfbyhs** [reply]
After the plugin failed to work, I tried running

**Comment by Mattmjs** [comment]
That was a ffprobe of the master manifest, I used a stream manifest
directly in ffmpeg without those extra parameters and it worked without any
issues.

**Comment by GrampaWildWilly** [comment]
This one is rather strange.  I haven't seen anything like this before.  I went to the URL @Mattmjs Matt provided.  I found the HLS master manifest in the Network Monitor, same as always.  I ran ffprobe on it:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15389631/ffprobe.txt)

You can see there were many corrupt packets there, which made me quite pessimistic about this one.  Once ffprobe was done reporting all those errors, it finally gave its usual report.  You'll note that every Stream looks bad except 0:6 & 0:7.  So I ran ffmpeg supplying -map 0:6 -map 0:7.  That failed to download anything.  Looking carefully at the ffprobe that is embedded in ffmpeg, I noticed that the one good Program wasn't Program 3, Streams 0:6 & 0:7.  It was some other Program.  So between the time I ran ffprobe on that manifest & the time I ran ffmpeg on that exact same manifest, the Programs & their Streams were no longer laid out the same way.

I picked the 2 Stream identifiers that my first run of ffmpeg said were the 1280x720 video & ran ffmpeg with those -map parameters.  It failed, too.  Reason?  The Programs had shuffled around again.

Well, there's only 1 Program in this master manifest that seems to actually contain program material.  So I quit fighting city hall & omitted -map parameters.  That ran:

[Test Video mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15389687/Test.Video.mp4.Log.txt)

You'll note that once again, the pattern of the Programs doesn't match what is shown in the ffprobe log above.

You'll also note that the download speed is rather putrid, just barely 250,000 bytes per second.  No doubt the download speed is severely hampered by the high volume of corrupt packets.

You'll also note that the duration of this item as detected by ffprobe & ffmpeg is N/A.  In other words, it's a livestream.  So after I had recorded about 5 minutes of content, I just gave ffmpeg a q.  You can see what it recorded at the bottom of that ffmpeg log.  The file it recorded was a bit over 5 minutes long.  It played just fine in VLC.  It's some kind of cartoon & everybody is speaking some language I don't understand, I'm guessing Japanese.  But it downloaded just fine & during playback in VLC my sampling of it showed both video & audio right to the end.

I say VDH might have no way to download this.  If the Stream assignments keep shifting around, how in the world are you supposed to program for that?  If you are to succeed at all on this content, you are probably going to have to handle it differently than anything else you've programmed for.  Of the 5 Programs in this item, 4 are completely corrupt, & 1 is at least recognizable, although it is also full of corruption.  In addition, the livestream is full, absolutely plagued by corrupt packets.

My take is that this one may simply not be worth adapting VDH for.  But Paul, you know the code & I've never seen it, so your analysis is far more valid than mine.

**Reply by GrampaWildWilly** [reply]
I believe there are several techniques that web sites can use to broadcast livestreams.  I'm going to describe 2 I know of.  At least, I think this is how they work.  I might be wrong about some of these details, but this is the current state of my understanding.  If I have some of this wrong, I'm sure somebody reading this will correct me.  I assume there are many more techniques in addition to these.  The web is anarchic about multimedia content so I assume that is also the case for livestreams.  These are just 2 I can describe.

First.  The web site publishes a master manifest located at some URL.  That manifest contains references to stream manifests at various resolutions covering some period of time, maybe seconds, maybe minutes, I don't really know.  The period is probably different for each web site.  When that time period has elapsed & all of the livestream described by that master manifest has been transmitted, the web site somehow chains to a new master manifest at a new URL.  The livestream then picks up with the new master manifest, which stays in use for however long applies in its case.  And so on, chaining from master manifest to master manifest, each one located at a new URL, until the livestream goes offline.  The end of the livestream is indicated by a failure to locate a new master manifest, or simply by completing whatever the last master manifest pointed to.

Second.  The web site publishes a master manifest located at some URL.  That manifest contains references to stream manifests at various resolutions covering some period of time, maybe seconds, maybe minutes, I don't really know.  The period is probably different for each web site.  When that time period has elapsed & all of the livestream described by that master manifest has been transmitted, the web site REPLACES the contents of the master manifest with new information to cover the next interval of the livestream.  The URL of the master manifest does not change, but the contents of the master manifest change periodically to control each succeeding segment of the livestream.  When the livestream ends, the web site either stops updating the master manifest, something the client software is supposed to recognize after some defined delay, or the web site removes the master manifest & the client software must recognize the not-found condition as the end of the livestream.

I am guessing that the cartoon web site uses something like the second technique.  That's why I kept seeing a different pattern of Programs being reported by ffprobe executed directly by me & by ffprobe executed by ffmpeg.  The software wasn't being capricious.  The web site was updating the master manifest.  What I think is somewhat capricious is that the one valid Program in the livestream wasn't always in the same place in the master manifest.  That seems like something the web site was doing on purpose to make life difficult for downloaders.  Except ffmpeg can be impervious to that sort of silliness, as my successful test showed.  So it's just annoying.

Maybe.  Subject to confirmation by somebody who knows more than I do.

**Reply by tyfbyhs** [reply]
So the current situation is probably that the website has restricted my downloads?

**Reply by GrampaWildWilly** [reply]
Not correct.  I downloaded stuff from there no problem.  The web site is not restricting anything.  It's just that VDH at the moment doesn't know what to do with what it's seeing.  My example above shows that you can quite easily record as much as you'd like using ffmpeg.  There is a way to tell ffmpeg to rewind to the beginning of a livestream & record it from the beginning, even if you joined the livestream late.  There is a caveat to that.  The web site may not keep the entirety of the livestream all the way back to its start.  It might keep only an hour or two or some other duration.  The point is, though, that ffmpeg can record from a point in the past, past being defined as before the moment you  tune in to the livestream.

So no, the web site isn't restricting anything.  You just need to learn how to use ffmpeg.  Do a Google search with "vdh ffmpeg" as your search key.  Take a look at the very first search result.  If you are motivated to expand your horizons & learn something new, ffmpeg will do a perfectly acceptable job of recording content from this cartoon web site.  I did it.  Anybody can do it.  That includes you.

I should make it clear that I do not work for VDH.  I'm just another user.

**Reply by tyfbyhs** [reply]
I may not be able to study much, my English is not very good and I can only barely communicate through a translator

**Reply by tyfbyhs** [reply]
Can you teach me how to do it?

**Reply by GrampaWildWilly** [reply]
I already have.  Do that Google search I mentioned above.  You will find the place where I already did the teaching.  I am not going to fill up space in this forum with a duplication of what I already did in our old forum.  I appreciate the problems imposed by a language barrier.  But there is no substitute for spending the difficult time it takes to learn something new.  It's up to you how hard you want to work at this.

**Reply by tyfbyhs** [reply]
OK, thanks for your help

**Reply by tyfbyhs** [reply]
I didn't find the tutorial you mentioned

**Reply by tyfbyhs** [reply]
{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://live.fc2.com/648390/",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 00000277d4a7a940] Error opening input: Invalid data found when processing input\r\nError opening input file https://us-west-1-media.live.fc2.com/a/stream/648390/32/playlist?c=B0ZWzZ_7dyWeeEGcg0Kze&d=NXbROY-0jjDsE8CRVKbFeOW1n2HDQz5UWPollASvDCycmS3t8OPYhGOBQC0Nz1S72-4IoPZs2Pa8EDKU9zOCR6hwmGyRHoFcCvIbPMv_AnM.\r\nError opening input files: Invalid data found when processing input\r\n",
    "id": "coapp_failure_ded0c5cf-47b7-4be3-ade6-27406d8e40b2",
    "downloadable_id": "downloadable_7500537878544806"
  }
}

**Reply by tyfbyhs** [reply]
It shows where there is an error but I won't fix it

**Reply by GrampaWildWilly** [reply]
Impossible.

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6f113e27-5140-4c86-a963-da8a021dbc8e)

**Reply by tyfbyhs** [reply]
How to deal with this error report?

**Reply by GrampaWildWilly** [reply]
Here's how you deal with it.  You wait patiently while the VDH developers deal with the issue.  That may take a day or two.  That may take a year or two.  The developers may decide not to correct this problem.  I don't know because, like I said earlier, I don't work for VDH.  So you can choose to wait, possibly in vain, for a fix to VDH.  Or you can learn how to use ffmpeg & get whatever content you please off this web site.  And you can get it right now.

The choice is yours.

**Reply by tyfbyhs** [reply]
Please forgive my stupidity, I still can't do it
